HW05.Rmd

```diff
@@ -0,0 +1,124 @@
+---
+title: "HW05"
+author: "Brock"
+date: "8/3/2020"
+output: html_document
+---
+#Goal: Tidy and join data :)
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+library("tidyverse") #Tidyverse functionalities
+library("knitr") #Nice outputs for tibbles/ dataframes
+library("here") #Pathing for importing data
+
+```
+
+Data importing is made easier with the here() function from the "here" package.
+
+```{r Data import, message = FALSE}
+dad_mom <- read_delim("dad_mom.txt", delim = "\t")
+
+auc_1 <- read_csv(here("./CTRP_files/AUC_1.csv")) #here() outputs path of current project directory; improved readability vs "C:/BLAH/BLAH/BLAH/BLAH/etc"
+
+auc_2 <- read_csv(here("./CTRP_files/AUC_2.csv"))
+
+cancer_cell_line_info <- read_csv(here("./CTRP_files/cancer_cell_line_info.csv"))
+
+compound_info <- read_csv(here("./CTRP_files/compound_info.csv"))
+
+experiment_info <- read_csv(here("./CTRP_files/Experiment_info.csv")) #here() is MVP
+```
+
+I was stuck on this for quite awhile. I wanted to find a way to combine the name_ income_ variables into a single column, but I couldn't think of a great way to do that. I look forward to reviewing the code for this assignment to see what people came up with!
+
+```{r tidy family data}
+tidy_family <- dad_mom %>%
+  gather(key = "key", value = "value", -fam_id, -name_dad, -name_mom) %>% #Gathers income and names of parents, keeps fam_id separate
+  mutate("parent_name" = c(name_dad[1:3], name_mom[1:3])) %>% #I couldn't think of a better way to do this
```

Comment:
I hadn't thought of that solution to combine the two columns! What I ended up doing was directly creating new columns in gather() when combining the mom and dad "income" and "name" columns.